File: Java/src/covariance/chemistry/CovAtom.java
package covariance.chemistry;

import covariance.abs.AbsGraph;
import covariance.abs.AbsVertex;

public class CovAtom extends AbsVertex
{
    public CovAtom(Integer label)
    {
        super(label);
    }

    public void setGraph(CovMolecule m)
    {
        super.setGraph(m);
    }

    @Override
    public void setGraph(AbsGraph g)
    {
        try
        {
            setGraph((CovMolecule) g);
        }
        catch (Exception e)
        {
            e.printStackTrace();
        }
    }
}

File: scala/src/covariance/chemistry/CovAtom.scala
package covariance.chemistry

import covariance.abs.AbsGraph
import covariance.abs.AbsVertex

class CovAtom(label:Integer) extends AbsVertex(label) {

  def setGraph(m: CovMolecule): Unit = {
    super.setGraph(m)
  }

  override def setGraph(g: AbsGraph): Unit = {
    try {
      setGraph(g.asInstanceOf[CovMolecule])
    } catch {
      case e: Exception =>
        e.printStackTrace()

    }
  }

}
